guides/keras_cv/cut_mix_mix_up_and_rand_augment.py

```diff
@@ -0,0 +1,396 @@
+"""
+Title: CutMix, MixUp, and RandAugment image augmentation with KerasCV
```

Comment:
@LukeWood 
**IMHO**, I think the guidebook regarding the augmentation with the new `keras_cv` can be much more appealing. Highlighting cutmix/mixup/rand_aug may not be new to know for most of the audience, as practitioners use these most of the time with image classification tasks. Maybe, demonstrating these for the detection task, or showing execution benchmark, etc would be catchy.  Additionally, it can be shown how conveniently a user can write his own augmentation layer in `keras_cv`.

Reply:
We aren't exposing BaseImageAugmentationLayer as an API from Keras yet.  Perhaps we will expose it from KerasCV at some point, but not until we add at least bbox and segm support.